server/lib/nl/detection/date.py

```diff
@@ -103,6 +105,15 @@ def parse_date(query: str, ctr: Counters) -> DateClassificationAttributes:
       dates.append(Date(_LAST_YEARS_PREP, year, year_span=1))
       trigger_strings.append(query[match.start():match.end()])
 
+  for pattern in YEARS_AGO_RE:
```

Comment:
To confirm, can each of these patterns be subset of each other? and thus need an ordered if else check?

Reply:
I don't think these patterns could be subsets of each other? also currently, if a query matches to more than one pattern, we add all the matched dates to the DateClassificationAttributes, but only use the first one in the list. Definitely something we can discuss changing, but probably a separate issue?

Reply:
I think if we use only one, then we might order them from the longest match to shortest match or something like that.  But, agreed, separate concern...